File: bot/src/main/java/by/yayauheny/tutochkatgbot/dto/backend/NearestRestroomResponseDto.java
package by.yayauheny.tutochkatgbot.dto.backend;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

/**
 * Nearest restroom response DTO
 */
public record NearestRestroomResponseDto(
    @JsonProperty("id") UUID id,
    @JsonProperty("name") String name,
    @JsonProperty("address") String address,
    @JsonProperty("coordinates") LatLon coordinates,
    @JsonProperty("distanceMeters") Double distanceMeters,
    @JsonProperty("feeType") FeeType feeType,
    @JsonProperty("isOpen") Boolean isOpen,
    @JsonProperty("placeType") PlaceType placeType,
    @JsonProperty("building") BuildingResponseDto building,
    @JsonProperty("subwayStation") SubwayStationResponseDto subwayStation
) {}

File: shared/src/main/kotlin/by/yayauheny/shared/dto/NearestRestroomResponseDto.kt
package by.yayauheny.shared.dto

import by.yayauheny.shared.enums.FeeType
import by.yayauheny.shared.enums.PlaceType
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class NearestRestroomResponseDto(
    @Contextual val id: UUID,
    val name: String?,
    val address: String,
    val coordinates: LatLon,
    val distanceMeters: Double,
    val feeType: FeeType,
    val isOpen: Boolean? = null,
    val placeType: PlaceType? = null,
    val building: BuildingResponseDto? = null,
    val subwayStation: SubwayStationResponseDto? = null
)
